mypy/errors.py

```diff
@@ -798,54 +782,89 @@ def remove_duplicates(self, errors: List[ErrorTuple]) -> List[ErrorTuple]:
 
     def save_baseline(self, file: Path) -> None:
         """Create/update a file that stores all errors"""
+        if not self.all_errors:
+            if not file.exists():
+                return
+            file.unlink()
+            print("No errors, baseline file removed")
+            return
         if not file.parent.exists():
             file.parent.mkdir()
         json.dump(
             {
-                file: [
+                remove_path_prefix(file, self.ignore_prefix).replace("\\", "/"): [
                     {
                         "line": error.line,
                         "code": error.code and error.code.code,
-                        "message": error.message
-                    } for error in errors
+                        "message": error.message,
+                        "target": error.target,
+                    } for error in self.sort_messages(errors)
+                    # don't store reveal errors
+                    if error.code != codes.REVEAL
                 ]
-                for file, errors in filter_prefix(self.error_info_map).items()
+                for file, errors in self.all_errors.items()
             },
             file.open("w"),
             indent=2,
+            sort_keys=True,
         )
 
     def load_baseline(self, file: Path) -> bool:
-        """Load baseline errors from baseline file"""
+        """Load baseline errors from baseline file
+
+        :raises JSONDecodeError: if the file contains invalid JSON
+        """
 
         if not file.exists():
             return False
         self.baseline = json.load(file.open("r"))
         return True
 
-    def filter_baseline(self) -> None:
+    def filter_baseline(self, path: str) -> None:
         """Remove baseline errors from the error_info_map"""
-
-        self.all_errors = self.error_info_map.copy()
-        for file, errors in self.error_info_map.items():
-            baseline_errors = self.baseline.get(
-                Path(file).resolve().relative_to(Path.cwd()).as_posix())
-            if not baseline_errors:
+        if path not in self.error_info_map:
+            return
+        if path not in self.all_errors:
+            self.all_errors[path] = self.error_info_map[path]
+        baseline_errors = self.baseline.get(
+            remove_path_prefix(path, self.ignore_prefix).replace('\\', '/')
```

Comment:
isnt there a built in thing to handle the different path separators? if not since you're doing this in multiple places maybe turn it into a function